### PR TITLE
Remove DOM manipulation and set list in onAdd and onRemove

### PR DESCRIPTION
**Pull request by MartinMalinda**
Should close https://github.com/Alfred-Skyblue/vue-draggable-plus/issues/77

Maybe I've removed too much stuff? I am not sure what are the checks for `isRef`? Isn't it safe to assume the list will always be a `ref` ? Should the lib work with non reactive arrays (why?)?

### CODE REVIEW

File: src/useDraggable.ts [hunk 1]
```diff
@@ -138,24 +139,29 @@ export function useDraggable<T>(...args: any[]): UseDraggableReturn {
...
+    const item = fromList.value[evt.oldIndex!];
+    if (isUndefined(item)) return
+    const newList = [...list.value];
+    insertElement(newList, evt.newDraggableIndex!, item);
+    list.value = newList;
+  }
+
+  function getListFromEl(el: HTMLElement) {
+    const key = Object.keys(el).find(key => key.startsWith('Sortable'));
+    return key && (el as any)[key]?.__list;
```
Comment (MartinMalinda) [inline src/useDraggable.ts:153]: This is a little bit ugly, but I need to have access to the original list to get the exact same object, not a clone. Cloning removes any getter and setter logic, thus kills reactivity.

Comment (MartinMalinda) [reply src/useDraggable.ts:153]: But at least with this approach we rely on native Vue rerender instead of DOM manipulation?

**Comment by Alfred-Skyblue** [comment]
If that's the case, we are changing the reference to the original array. Essentially, we are performing operations like adding, deleting, or moving elements within the original array. If we forcefully change the reference to the array, it could potentially lead to unnecessary issues, such as watch deep.

**Comment by MartinMalinda** [comment]
I'm very frequently doing `list.value = [...newList.value]` without issues. This triggers the setter on the `ref` and reactivity works just fine. I think what's important is not to break the child objects reactivity by cloning.

Besides the code is already changing the reference to the original array on the most common operation, changing order:
https://github.com/Alfred-Skyblue/vue-draggable-plus/blob/main/src/useDraggable.ts#L175

**Comment by Alfred-Skyblue** [comment]
> I am not sure what are the checks for isRef? Isn't it safe to assume the list will always be a ref ? Should the lib work with non reactive arrays (why?)?

Because ref is unpacked in the directive, isRef is used in useDraggable

> But at least with this approach we rely on native Vue rerender instead of DOM manipulation 🤔

Manual manipulation of the DOM is necessary to trigger the animation effect.

**Comment by MartinMalinda** [comment]
Thanks for the reply, I'll close this and proceed with forking.

**Pull request closed without merging**